### ISSUE TITLE
`chunks_exact_to_as_chunks` suggest invalid const generic parameter expression

### ISSUE THREAD

**Issue by RunDevelopment** [rust-lang/rust-clippy#17314]
### Summary

The recently added `chunks_exact_to_as_chunks` lint (#16931) produces false positives when the constant expression is derived from a const generic parameter. 

Example:

```rs
fn foo<const N: usize>(buf: &[u8]) {
    let _ = buf.chunks_exact(N * 2);
}
```

This currently cannot be expressed using `as_chunks` in stable Rust due to limitations around const generics. E.g. compiling `buf.as_chunks::<{ N * 2 }>()` gives:

```
error: generic parameters may not be used in const operations                                                                                                       
   --> src\main.rs:7:31
    |
388 |     let _ = buf.as_chunks::<{ N * 2 }>();
    |                               ^ cannot perform const operation using `N`
    |
    = help: const parameters may only be used as standalone arguments here, i.e. `N`
    = help: add `#![feature(generic_const_exprs)]` to allow generic const expressions
```

### Lint Name

chunks_exact_to_as_chunks

### Reproducer

I tried this code:

```rust
fn foo<const N: usize>(buf: &[u8]) {
    let _ = buf.chunks_exact(N * 2);
}
```

I saw this happen:

```
warning: using `chunks_exact` with a constant chunk size
 --> src/main.rs:7:17
  |
7 |     let _ = buf.chunks_exact(N * 2);
  |                 ^^^^^^^^^^^^^^^^^^^
  |
help: consider using `as_chunks::<N * 2>()` instead
 --> src/main.rs:7:17
  |
7 |     let _ = buf.chunks_exact(N * 2);
  |                 ^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#chunks_exact_to_as_chunks
  = note: `#[warn(clippy::chunks_exact_to_as_chunks)]` on by default

```

I expected to see this happen:

No warning. 

Also note that the help code snippet `as_chunks::<N * 2>()` is not syntactically valid Rust.

### Version

```text
rustc 1.98.0-nightly (13f1859f2 2026-06-27)
binary: rustc
commit-hash: 13f1859f2faf97a15664e655624baa7417fdc100
commit-date: 2026-06-27
host: x86_64-pc-windows-msvc
release: 1.98.0-nightly
LLVM version: 22.1.8
```
